src/screens/walletConnect/WalletConnect2Context.tsx

```diff
@@ -286,26 +291,34 @@ export const WalletConnect2Provider = ({
     }
   }
 
-  const onDisconnectSession = async (session: SessionStruct) => {
-    try {
-      const web3wallet = await createWeb3Wallet()
-      await web3wallet.disconnectSession({
-        topic: session.topic,
-        reason: getSdkError('USER_DISCONNECTED'),
-      })
-      setSessions(prevSessions =>
-        prevSessions.filter(prevSession => prevSession.topic !== session.topic),
-      )
-    } catch (err) {
-      // @TODO handle error disconnecting
-      console.log('WC2.0 error disconnect', err)
-    }
-  }
+  const onDisconnectSession = useCallback(
+    async (session: SessionStruct) => {
+      if (web3wallet) {
+        try {
+          await web3wallet.disconnectSession({
+            topic: session.topic,
+            reason: getSdkError('USER_DISCONNECTED'),
+          })
+          setSessions(prevSessions =>
+            prevSessions.filter(
+              prevSession => prevSession.topic !== session.topic,
+            ),
+          )
+        } catch (err) {
+          // @TODO handle error disconnecting
+          console.log('WC2.0 error disconnect', err)
```

Comment:
If there is an error disconnecting, remove the session locally for the user. These errors happen on the client side and the user can reconnect to the dapp if needed.

Reply:
I did not make that TODO but if necessary I can do that here or in the separate ticket because this one was about being able to delete the session